### PR TITLE
feat(forms): add high-density support to Select

### PR DESCRIPTION
**Pull request by cf-remylenoir**
# Purpose of PR

Enables high-density support for the `Select` component (medium size only).

Ticket: https://contentful.atlassian.net/browse/CFCX-958

Docs: https://forma-36-git-feat-select-high-density-support.colorfuldemo.com/components/select#density-support
Storybook: https://5fd1dda724cc620021ace8c5-znbenjvdei.chromatic.com/?path=/story/form-elements-select--with-density-support

with-ellipsis

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/main/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

**Comment by cf-remylenoir** [comment]
@damann the text gets slightly cut with the current `height/padding` values (see the bottom of the `p`).

To keep the `height` in control (32px), I suggest going one more level down with the `padding`.

| Value | Outcome |
|--------|--------|
| Top/Bottom: spacingXs (8px) | xs |
| Top/Bottom: spacing2Xs (2px) | 2xs |

File: packages/components/forms/src/Select/Select.styles.ts [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
           height: '32px',
+          fontSize: tokens.fontSizeM,
+          lineHeight: tokens.lineHeightM,
+          borderRadius: tokens.borderRadiusMedium,
         }
       : {
-          padding: `10px ${tokens.spacingL} 10px ${tokens.spacingS}`,
-          height: '40px',
+          padding: isHighDensity
+            ? `${tokens.spacingXs} ${tokens.spacingL} ${tokens.spacingXs} ${tokens.spacingXs}`
```
Comment (Lelith) [inline packages/components/forms/src/Select/Select.styles.ts:17]: I am a little concerned about the amount of inline ifs for this one variable. Would it not be easier to read and maintain to switch the whole class or write a "getStyles" sort of function? So we would have one if and avoid touching each and every single property

File: packages/components/forms/src/Select/Select.styles.ts [hunk 2]
```diff
@@ -28,6 +43,7 @@ export function getSelectStyles({ isInvalid, isDisabled, size }) {
     outline: 'none',
     border: `1px solid ${tokens.gray300}`,
     cursor: 'pointer',
+    textOverflow: 'ellipsis',
```
Comment (cf-remylenoir) [inline packages/components/forms/src/Select/Select.styles.ts:46]: The `text-overflow` handling was missing, resulting in cut text when too long

| Before | After |
|--------|--------|
|  no-ellipsis | with-ellipsis |

File: packages/components/forms/src/Select/Select.styles.ts (hunk 1 above)
Comment (cf-remylenoir) [reply packages/components/forms/src/Select/Select.styles.ts]: It makes sense, adjusted 👍

**Review by damann — APPROVED** [review]

**Comment by cf-remylenoir** [comment]
@Lelith please let me know if you are satisfied with the changes 🙂

**Pull request merged**